text/0000-inline-buildpack.md

```diff
@@ -0,0 +1,117 @@
+# Meta
+[meta]: #meta
+- Name: Inline Buildpacks
+- Start Date: 2020-06-11
+- Author(s): [Joe Kutner](https://github.com/jkutner)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a type of buildpack that can be defined in the same repo as the app it is used with.
+
+# Motivation
+[motivation]: #motivation
+
+One characteristic of `Dockerfile` that Buildpacks do not replace well is the ability to create a single, one-off file to quickly define a custom build process for an app or project. A `Dockerfile` may contain just a few lines of code, and a developer can generate a complete image. The only quick way to create a custom build for an app using a buildpack is by creating no fewer than three files: `buildpack.toml`, `bin/build`, and `bin/detect` (and likely storing those in a new repo).
+
+We seek to reduce the barrier to entry for buildpacks when a developer needs to quickly create a custom, one-off build step by introducing a new mechanism that will allow a buildpack to be defined inline with an app (and with fewer files).
+
+# What it is
+[what-it-is]: #what-it-is
+
+- **inline** - existing in the same repository
+
+The target personas for this RFC are the buildpack author and buildpack user who need to create a custom, one-off build or are just getting started with buildpacks and do not want the overhead of a new repo.
+
+We propose two new keys in the `[[build.buildpacks]]` table in `project.toml`:
+
+- `inline` - a script that will be run as the `bin/build` of the inline buildpack
+- `shell` - defines the shebang line of the script in `inline`
+
+(Note: there is no need to implement `bin/detect`. The detect phase will always pass because there's no foreseeable reason that a user would create an inline buildpack that they don't want to use)
+
+# How it Works
+[how-it-works]: #how-it-works
+
+When an entry in the `[[build.buildpacks]]` table contains an `inline` value, the lifecycle WILL NOT attempt to download the buildpack `id@version`. Instead, it will create an ephemeral buildpack, with the following:
```

Comment:
This makes it sounds like the lifecycle currently downloads missing buildpacks. It does not do this. Buildpacks must be present on the filesystem when the lifecycle is executed.

Reply:
Also I believe we have deliberately avoided have the lifecycle directly read `project.toml`, do we want the platform to create an ephemeral buildpack? I lean towards providing this information to the lifecycle in another format.

Reply:
oh yea, i just meant to put platform/pack here

Reply:
Fixed in c74c093